src/sentry/tasks/repository.py

```diff
@@ -0,0 +1,33 @@
+from sentry.deletions import default_manager
+from sentry.deletions.base import _delete_children
+from sentry.deletions.defaults.repository import _get_repository_child_relations
+from sentry.models.repository import Repository
+from sentry.tasks.base import instrumented_task
+
+
+@instrumented_task(
+    name="sentry.models.repository_cascade_delete_on_hide",
```

Comment:
I think you'll want `acks_late=True` on this task so that it retries until it completes. Long running tasks can get killed by deploys or k8s moving a pod around.